src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -253,81 +252,94 @@ private static CreatedStudyBasicInfos toCreatedStudyBasicInfos(StudyEntity entit
+                .map(StudyService::toCreatedStudyBasicInfos)
+                .sorted(Comparator.comparing(CreatedStudyBasicInfos::getCreationDate).reversed())
+                .collect(Collectors.toList());
     }
 
-    public Flux<CreatedStudyBasicInfos> getStudiesMetadata(List<UUID> uuids) {
-        return Flux.fromStream(() -> studyRepository.findAllById(uuids).stream().map(StudyService::toCreatedStudyBasicInfos));
+    public List<CreatedStudyBasicInfos> getStudiesMetadata(List<UUID> uuids) {
+        return studyRepository.findAllById(uuids).stream().map(StudyService::toCreatedStudyBasicInfos)
+                .collect(Collectors.toList());
     }
 
-    Flux<BasicStudyInfos> getStudiesCreationRequests() {
-        return Flux.fromStream(() -> studyCreationRequestRepository.findAll().stream())
-            .map(StudyService::toBasicStudyInfos)
-            .sort(Comparator.comparing(BasicStudyInfos::getCreationDate).reversed());
+    List<BasicStudyInfos> getStudiesCreationRequests() {
+        return studyCreationRequestRepository.findAll().stream()
+                .map(StudyService::toBasicStudyInfos)
+                .sorted(Comparator.comparing(BasicStudyInfos::getCreationDate).reversed()).collect(Collectors.toList());
     }
 
-    public Mono<BasicStudyInfos> createStudy(UUID caseUuid, String userId, UUID studyUuid) {
+    public BasicStudyInfos createStudy(UUID caseUuid, String userId, UUID studyUuid) {
         AtomicReference<Long> startTime = new AtomicReference<>();
-        return insertStudyCreationRequest(userId, studyUuid)
-                .doOnSubscribe(x -> startTime.set(System.nanoTime()))
-                .map(StudyService::toBasicStudyInfos)
-                .doOnSuccess(s -> Mono.zip(persistentStore(caseUuid, s.getId(), userId), getCaseFormat(caseUuid))
-                        .flatMap(t -> {
-                            LoadFlowParameters loadFlowParameters = LoadFlowParameters.load();
-                            return insertStudy(s.getId(), userId, t.getT1().getNetworkUuid(), t.getT1().getNetworkId(),
-                                    t.getT2(), caseUuid, false, toEntity(loadFlowParameters));
-                        })
-                        .subscribeOn(Schedulers.boundedElastic())
-                        .doOnError(throwable -> LOGGER.error(throwable.toString(), throwable))
-                        .doFinally(st -> {
-                            deleteStudyIfNotCreationInProgress(s.getId(), userId).subscribe();
-                            LOGGER.trace("Create study '{}' : {} seconds", s.getId(), TimeUnit.NANOSECONDS.toSeconds(System.nanoTime() - startTime.get()));
-                        })
-                        .subscribe()
-                );
+        startTime.set(System.nanoTime());
+        BasicStudyInfos basicStudyInfos = StudyService.toBasicStudyInfos(insertStudyCreationRequest(userId, studyUuid));
+        studyServerExecutionService.runAsync(() -> {
```

Comment:
Add a method for the final  study creation (to add later a `Transactional` annotation
`studyServerExecutionService.runAsync(() -> createStudy(caseUuid, basicStudyInfos));`

Reply:
fixed